beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/singlepeer/PeerSync.java

```diff
@@ -181,42 +164,12 @@ private SafeFuture<PeerSyncResult> executeSync(
               final SafeFuture<Void> readyForNextRequest =
                   asyncRunner.getDelayedFuture(NEXT_REQUEST_TIMEOUT);
 
-              final BlockAndBlobsSidecarMatcher blockAndBlobsSidecarMatcher =
-                  new BlockAndBlobsSidecarMatcher(
-                      blobsSidecarManager,
-                      this::storeBlobsSidecarAndImportBlock,
-                      this::importBlock);
-
-              final RpcResponseListener<SignedBeaconBlock> blockListener;
-              final SafeFuture<Void> blobsSidecarsRequest;
-
-              if (requestContext.areBlobsSidecarsRequired()) {
-                blockListener = blockAndBlobsSidecarMatcher::recordBlock;
-                LOG.debug(
-                    "Request {} blobs sidecars starting at {} from peer {}",
-                    count,
-                    ancestorStartSlot,
-                    peer.getId());
-                blobsSidecarsRequest =
-                    peer.requestBlobsSidecarsByRange(
-                        ancestorStartSlot, count, blockAndBlobsSidecarMatcher::recordBlobsSidecar);
-              } else {
-                blockListener = this::importBlock;
-                blobsSidecarsRequest = SafeFuture.COMPLETE;
-              }
-
               final PeerSyncBlockRequest request =
                   new PeerSyncBlockRequest(
-                      readyForNextRequest, ancestorStartSlot, count, blockListener);
-
-              return SafeFuture.allOfFailFast(
-                      peer.requestBlocksByRange(ancestorStartSlot, count, request),
-                      blobsSidecarsRequest)
-                  .thenApply(
-                      __ -> {
-                        blockAndBlobsSidecarMatcher.clearCache();
-                        return request;
-                      });
+                      readyForNextRequest, ancestorStartSlot, count, this::importBlock);
+
+              return peer.requestBlocksByRange(ancestorStartSlot, count, request)
+                  .thenApply(__ -> request);
```

Comment:
let's rename `request` -> `blockRequest` so it will match the following compose lambda param